packages/hardhat-test-helpers/src/utils.ts

```diff
@@ -0,0 +1,88 @@
+import type { EIP1193Provider } from "hardhat/types";
+
+import type { NumberLike } from "./types";
+
+async function checkIfHardhatNetwork(
+  provider: EIP1193Provider
+): Promise<boolean> {
+  try {
+    // we call a method that only exists in the hardhat network
+    await provider.request({
+      method: "hardhat_getAutomine",
+    });
+
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+export async function getHardhatProvider(): Promise<EIP1193Provider> {
```

Comment:
We should cache this method, otherwise each call to a helper will make one extra JSON-RPC call to check if the network is a Hardhat network.